### PR TITLE
Enhance "wrapping C libraries".

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
- Used `intptr_t` instead of `int` to avoid warnings of unsafe cast (and crashes since C++ consider casting from `void*` to `int` an error). 
- Moved the main implementation of the queue into the examples directory for testing.
- Removed the old examples files (they were not used anyway), all the equivalents are in `examples/tutorial/clibraries`.

### CODE REVIEW

**Comment by scoder** [comment]
Well, there is the "little" issue that MSVC lacks `stdint.h`, so it would be nice to have the example get along without it. Maybe `Py_ssize_t` would work as well?

Also, using `intptr_t` in the `cdef` method signatures is really unhelpful as the idea is to pass an `int` arrray as an additional C-level interface. Better restrict the specific integer types to casts and leave everything else with `int` to keep the external interface clean.

File: docs/src/tutorial/clibraries.rst [hunk 1]
```diff
@@ -574,12 +505,12 @@ predicate.  The API could look as follows::
...
      *  1 for accept
      */
-    typedef int (*predicate_func)(void* user_context, QueueValue data);
+    typedef intptr_t (*predicate_func)(void* user_context, QueueValue data);
 
     /* Pop values as long as the predicate evaluates to true for them,
      * returns -1 if the predicate failed with an error and 0 otherwise.
      */
-    int queue_pop_head_until(Queue *queue, predicate_func predicate,
+    intptr_t queue_pop_head_until(Queue *queue, predicate_func predicate,
```
Comment (scoder) [inline docs/src/tutorial/clibraries.rst]: Sorry, but this is really not what the C-API of the queue looks like.

**Comment by gabrieldemarmiesse** [comment]
It's true that it's a lot cleaner now. My experience with plain C is somewhat limited, as you can see. It's getter better with your reviews.

File: docs/examples/tutorial/clibraries/queue3.pyx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        for i in range(count):
+            if not cqueue.queue_push_tail(
+                    self._c_queue, <void*> values[i]):
+                raise MemoryError()
+
+    # The `cpdef` feature is obviously not available for the `extend()`
+    # method, as the method signature is incompatible with Python argument
+    # types (Python doesn't have pointers). However, we can make a method
+    # called `extend_python` instead that accepts an arbitrary Python iterable.
+    cpdef extend_python(self, values):
```
Comment (scoder) [inline docs/examples/tutorial/clibraries/queue3.pyx]: The naming of these two methods was intentional. The Python interface dominates the API of this class and should be consistent. `extend()` is the method name a Python user would expect. The additional C-interface should be a second level citizen wherever it cannot matche the Python interface.

Maybe we could rename the cdef `extend()` method to `extend_ints()`. That way, we could easily add an `extend_longs()` or `extend_chars()` method for convenience, as one would do in C APIs.

**Pull request merged**